components/scream/src/physics/p3/p3_main_impl.hpp

```diff
@@ -963,121 +919,123 @@ ::p3_main(
         &inv_dzq, &inv_rho, &ze_ice, &ze_rain, &prec, &rho,
         &rhofacr, &rhofaci, &acn, &qvs, &qvi, &sup, &supi,
-        &tmparr1, &inv_exner, &qtend_ignore, &ntend_ignore,
+        &tmparr1, &inv_exner, &diag_ze, &diag_vmi, &diag_di,
+        &pratot, &prctot, &qtend_ignore, &ntend_ignore
       });
 
     // Get single-column subviews of all inputs, shouldn't need any i-indexing
     // after this.
-    const auto opres             = util::subview(pres, i);
-    const auto odzq              = util::subview(dzq, i);
-    const auto oncnuc            = util::subview(ncnuc, i);
-    const auto onaai             = util::subview(naai, i);
-    const auto oqc_relvar        = util::subview(qc_relvar, i);
-    const auto opdel             = util::subview(pdel, i);
-    const auto oexner            = util::subview(exner, i);
-    const auto oicldm            = util::subview(icldm, i);
-    const auto olcldm            = util::subview(lcldm, i);
-    const auto orcldm            = util::subview(rcldm, i);
-    const auto ocol_location     = util::subview(col_location, i);
-    const auto oqc               = util::subview(qc, i);
-    const auto onc               = util::subview(nc, i);
-    const auto oqr               = util::subview(qr, i);
-    const auto onr               = util::subview(nr, i);
-    const auto oqitot            = util::subview(qitot, i);
-    const auto oqirim            = util::subview(qirim, i);
-    const auto onitot            = util::subview(nitot, i);
-    const auto obirim            = util::subview(birim, i);
-    const auto oqv               = util::subview(qv, i);
-    const auto oth               = util::subview(th, i);
-    const auto odiag_ze          = util::subview(diag_ze, i);
-    const auto odiag_effc        = util::subview(diag_effc, i);
-    const auto odiag_effi        = util::subview(diag_effi, i);
-    const auto odiag_vmi         = util::subview(diag_vmi, i);
-    const auto odiag_di          = util::subview(diag_di, i);
-    const auto odiag_rhoi        = util::subview(diag_rhoi, i);
-    const auto omu_c             = util::subview(mu_c, i);
-    const auto olamc             = util::subview(lamc, i);
-    const auto ocmeiout          = util::subview(cmeiout, i);
-    const auto oprain            = util::subview(prain, i);
-    const auto onevapr           = util::subview(nevapr, i);
-    const auto oprer_evap        = util::subview(prer_evap, i);
-    const auto orflx             = util::subview(rflx, i);
-    const auto osflx             = util::subview(sflx, i);
-    const auto opratot           = util::subview(pratot, i);
-    const auto oprctot           = util::subview(prctot, i);
-    const auto oliq_ice_exchange = util::subview(liq_ice_exchange, i);
-    const auto ovap_liq_exchange = util::subview(vap_liq_exchange, i);
-    const auto ovap_ice_exchange = util::subview(vap_ice_exchange, i);
+    const auto opres             = util::subview(diagnostic_inputs.pres, i);
```

Comment:
Out of curiosity, what does the "o" prefix stand for? I can't think of any o-word that express locality to a single column. My actual first guess was 'output', but since all the subviews have it, it's not that.

In case you are not a fan of all those oxyz names, notice that, since the inputs are now inside a struct, the names _without_ the prefix are again available for use.

Reply:
@jgfouca , can you comment on this? I've just been treading lightly here, and it'd be interesting to hear what you intended with this prefix.

Reply:
I think the goal was just to be able to have 2 different variables with basically the same name. Maybe o stands for One-column or lOcal? In any case, using the fact that inputs are in a struct to get rid of those mysterious "o"s sounds like a good idea to me. But we should hear @jgfouca 's opinion before following my brash judgement.

Reply:
@bartgol , the "o" prefix means it's a single column subview of a 2D view